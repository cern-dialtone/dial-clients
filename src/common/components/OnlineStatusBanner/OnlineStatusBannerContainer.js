import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { setOnlineStatus } from 'settings/actions/settings';
import OnlineStatusBanner from './OnlineStatusBanner';

function mapStateToProps({ settings, auth }) {
  return {
    onlineStatus: settings.settings.onlineStatus,
    authError: auth.error
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      setOnlineStatus
    },
    dispatch
  );
}

export const ConnectedOnlineStatusBanner = connect(
  mapStateToProps,
  mapDispatchToProps
)(OnlineStatusBanner);

export default ConnectedOnlineStatusBanner;
