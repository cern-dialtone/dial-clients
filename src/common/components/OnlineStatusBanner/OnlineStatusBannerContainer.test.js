import React from 'react';
import { render, waitForElement } from '@testing-library/react';

import { Provider } from 'react-redux';

import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';

import { createHashHistory } from 'history';
import { routerMiddleware } from 'connected-react-router';

import OnlineStatusBannerContainer from './OnlineStatusBannerContainer';

const history = createHashHistory();
const middlewares = [thunk, routerMiddleware(history)];
const mockStore = configureMockStore(middlewares);

describe('OnlineStatusBannerContainer test', () => {
  let store;
  let initialState;

  beforeEach(() => {
    initialState = {
      call: {
        error: undefined
      },
      connection: {
        connected: true
      },
      numbers: {
        error: undefined
      },
      auth: {
        error: undefined
      },
      settings: {
        settings: {
          onlineStatus: true
        }
      }
    };
    store = mockStore(initialState);
    store.dispatch = jest.fn();
  });

  it('renders component with online status', async () => {
    const { container, asFragment } = render(
      <Provider store={store}>
        <OnlineStatusBannerContainer />
      </Provider>
    );
    expect(container.firstChild).toBeNull();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders component with offline status', async () => {
    initialState.settings.settings.onlineStatus = false;

    const { getByText, asFragment } = render(
      <Provider store={store}>
        <OnlineStatusBannerContainer />
      </Provider>
    );
  });
});
