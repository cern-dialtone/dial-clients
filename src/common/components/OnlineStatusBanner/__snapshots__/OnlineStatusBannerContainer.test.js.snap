// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnlineStatusBannerContainer test renders component with online status 1`] = `<DocumentFragment />`;
